crates/host/src/wasmbus/mod.rs

```diff
@@ -3210,38 +3352,69 @@ impl Host {
         Ok((config, secrets))
     }
 
+    /// Validates that the provided configuration names exist in the store and are valid.
+    ///
+    /// For any configuration that starts with `SECRET_`, the configuration is expected to be a secret reference.
+    /// For any other configuration, the configuration is expected to be a [`HashMap<String, String>`].
+    async fn validate_config(&self, config_names: Vec<String>) -> anyhow::Result<()> {
+        let config_store = self.config_data.clone();
+        let validation_errors = futures::future::join_all(
+            config_names.iter().map(|config_name| {
+                let config_store = config_store.clone();
+                async move {
+                    match config_store.get(config_name).await {
```

Comment:
Since we validate the data on storage and validate when actually getting the data for the component or provider, this should just check that the key exists and return the errors described in the last 2 cases of the match block